lib/SPIRV/SPIRVUtil.cpp

```diff
@@ -2185,12 +2186,27 @@ bool postProcessBuiltinReturningStruct(Function *F) {
                                      F->getReturnType());
       NewF->addParamAttr(0, SretAttr);
       NewF->setCallingConv(F->getCallingConv());
+      SmallVector<User *> Users(CI->users());
+      Value *A = nullptr;
+      for (auto *U : Users) {
+        if (auto *SI = dyn_cast<StoreInst>(U)) {
+          A = SI->getPointerOperand();
+          InstToRemove.push_back(SI);
+          break;
+        }
+      }
+      if (!A) {
+        A = Builder.CreateAlloca(F->getReturnType());
+      }
       auto Args = getArguments(CI);
-      Args.insert(Args.begin(), ST->getPointerOperand());
-      auto *NewCI = CallInst::Create(NewF, Args, CI->getName(), CI);
+      Args.insert(Args.begin(), A);
+      CallInst *NewCI = Builder.CreateCall(NewF, Args, CI->getName());
```

Comment:
Wait a sec. NewF here is void with sret, meanwhile Args were set for a previous instance of CI aka without sret parameter. That is incorrect.

Reply:
The problem here isn't about sret parameter. In llvm IR type system `addrspace` is an integral part of the type so `ptr != ptr addrspace(x)`. Indeed, this code had an error which is now fixed.